.script/tests/KqlvalidationsTests/JsonFilesTestData/DataConnectorFilesLoader.cs

```diff
@@ -25,53 +24,84 @@ protected override List<string> GetDirectoryPaths()
         public override List<string> GetFilesNames()
         {
             List<string> validFiles = new List<string>();
+
             try
             {
                 var directoryPaths = GetDirectoryPaths();
+                int prNumber = 0;
+                int.TryParse(System.Environment.GetEnvironmentVariable("PRNUM"), out prNumber);
+                //assign pr number to debug with a pr
+                //prNumber=8414;
+                var basePath = Utils.GetTestDirectory(TestFolderDepth);
+                var prFilesListModified = new List<string>();
 
-                return directoryPaths.Aggregate(new List<string>(), (accumulator, directoryPath) =>
+                if (prNumber != 0)
```

Comment:
Can we add condition if(prNumber == 0) then return.

Reply:
if(prNumber == 0) in this case we are loading all files, this is to make sure that if we are unable to fetch the PR number or unable to fetch the files from PR then we are making sure that validations run on all files.